lib/matplotlib/tests/test_preprocess_data.py

```diff
@@ -121,88 +102,83 @@ def real_func(x, y):
     def func(*args, **kwargs):
         real_func(**kwargs)
 
-    def f():
-        func(None, x="a", y="b")
-
     # This sets a label although the function can't handle it.
-    assert_raises(TypeError, f)
+    with pytest.raises(TypeError):
+        func(None, x="a", y="b")
 
 
-def test_function_call_without_data():
+@pytest.mark.parametrize('func', all_funcs, ids=all_func_ids)
+def test_function_call_without_data(func):
     """test without data -> no replacements"""
-    for func in all_funcs:
-        assert_equal(func(None, "x", "y"),
-                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: None")
-        assert_equal(func(None, x="x", y="y"),
-                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: None")
-        assert_equal(func(None, "x", "y", label=""),
-                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: ")
-        assert_equal(func(None, "x", "y", label="text"),
-                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: text")
-        assert_equal(func(None, x="x", y="y", label=""),
-                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: ")
-        assert_equal(func(None, x="x", y="y", label="text"),
-                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: text")
-
-
-def test_function_call_with_dict_data():
+    assert (func(None, "x", "y") ==
+            "x: ['x'], y: ['y'], ls: x, w: xyz, label: None")
+    assert (func(None, x="x", y="y") ==
+            "x: ['x'], y: ['y'], ls: x, w: xyz, label: None")
+    assert (func(None, "x", "y", label="") ==
+            "x: ['x'], y: ['y'], ls: x, w: xyz, label: ")
+    assert (func(None, "x", "y", label="text") ==
+            "x: ['x'], y: ['y'], ls: x, w: xyz, label: text")
+    assert (func(None, x="x", y="y", label="") ==
+            "x: ['x'], y: ['y'], ls: x, w: xyz, label: ")
+    assert (func(None, x="x", y="y", label="text") ==
+            "x: ['x'], y: ['y'], ls: x, w: xyz, label: text")
+
+
+@pytest.mark.parametrize('func', all_funcs, ids=all_func_ids)
```

Comment:
I really dislike this. It makes the tests much harder to read for no good reason.